packages/@aws-cdk/aws-eks/lib/fargate-profile.ts

```diff
@@ -145,6 +145,9 @@ export class FargateProfile extends Construct implements ITaggable {
       managedPolicies: [ iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSFargatePodExecutionRolePolicy') ],
     });
 
+    const creationRoleArn = props.cluster._getKubectlCreationRoleArn();
```

Comment:
There is an `iam.Role` object hiding there behind this method. Let's change it to return it instead of its ARN so we don't need `fromRoleArn` here.

Reply:
> There is an `iam.Role` object hiding there behind this method. Let's change it to return it instead of its ARN so we don't need `fromRoleArn` here.

I didn't get it. Do you mean to expose the creationRole from the custom resource so we can access it?

https://github.com/aws/aws-cdk/blob/03e85eb5629f87b34005422dfeb367d5581e85e8/packages/%40aws-cdk/aws-eks/lib/cluster-resource.ts#L32

Reply:
I think so... I need to dive a little deeper.